### PR TITLE
`thread::current_cpu` addition.

### PR DESCRIPTION
**Pull request by devnexen**
Allows to get the cpu id from the current running thread.

### CODE REVIEW

File: library/std/src/sys/unix/thread.rs [hunk 1]
```diff
@@ -385,6 +385,30 @@ pub fn available_parallelism() -> io::Result<NonZeroUsize> {
...
+    } else {
+        Err(io::const_io_error!(io::ErrorKind::NotFound, "The current cpu could not be fetched"))
+    }
+}
+
+#[cfg(not(any(target_os = "linux", target_vendor = "apple")))]
+pub fn current_cpu() -> io::Result<u32> {
+    Err(io::const_io_error!(
+        io::ErrorKind::Unsupported,
+        "The current cpu could is unsupported on this platform"
```
Comment (bjorn3) [inline library/std/src/sys/unix/thread.rs:408]: Can you rephrase this message to have correct grammar?

File: library/std/src/thread/mod.rs [hunk 2]
```diff
@@ -1652,3 +1652,22 @@ fn _assert_sync_and_send() {
...
+/// Fetches the running cpu from the current thread
+///
+/// # Examples
+///
+/// ```
+/// # #![allow(dead_code)]
+/// use std::{io, thread};
+///
+/// fn main() -> io::Result<()> {
+///     let cpu = thread::current_cpu()?.get();
```
Comment (bjorn3) [inline library/std/src/thread/mod.rs:1665]: Why the `.get()` here?

File: library/std/src/thread/mod.rs [hunk 3]
```diff
@@ -1652,3 +1652,22 @@ fn _assert_sync_and_send() {
...
+///     assert!(cpu >= 0u32);
```
Comment (bjorn3) [inline library/std/src/thread/mod.rs:1666]: This assertion trivially holds for unsigned integers.

**Comment by nagisa** [comment]
The most likely outcome for this PR is a request for this API to be vetted in an external crate before inclusion in the standard library can be considered.

**Comment by Mark-Simulacrum** [comment]
This needs a libs-api ACP -- I'll post the message that triagebot should have (had a bug last couple days):

Hey! It looks like you've submitted a new PR for the library teams! If this PR contains changes to any `rust-lang/rust` public library APIs then please comment with `@rustbot label +T-libs-api -T-libs` to tag it appropriately. If this PR contains changes to any unstable APIs please edit the PR description to add a link to the relevant [API Change Proposal](https://std-dev-guide.rust-lang.org/feature-lifecycle/api-change-proposals.html) or [create one](https://github.com/rust-lang/libs-team/issues/new?assignees=&labels=api-change-proposal%2C+T-libs-api&template=api-change-proposal.md&title=%28My+API+Change+Proposal%29) if you haven't already. If you're unsure where your change falls no worries, just leave it as is and the reviewer will take a look and make a decision to forward on if necessary.

Examples of `T-libs-api` changes:
* Stabilizing library features
* Introducing insta-stable changes such as new implementations of existing stable traits on existing stable types
* Introducing new or changing existing unstable library APIs (excluding permanently unstable features / features without a tracking issue)
* Changing public documentation in ways that create new stability guarantees
* Changing observable runtime behavior of library APIs

**Comment by the8472** [comment]
Closing this since the ACP was rejected in favor of having this functionality as an external crate.
